### DISCUSSION TITLE
Unable to create file /var/www/.composer.json because Permission denied

### DISCUSSION THREAD

**Discussion by medinabcarlos** [markshust/docker-magento#973] [Q&A]
When I run the script, I end up getting the following message:
[ErrorException] otouch(): Unable to create file /var/www/.composer.json because Permission denied

Does it happen to anyone else? how have you solved it? i tried with in a virtual machines with:
Ubuntu 22.04.02 LTS (VM)
4 CPU
SSD 200GB
8 GB RAM

**Comment by YevhenZvieriev** [comment]
Hi, @medinabcarlos 

This occurs because the owner of the `~/.composer` folder is root. When it's copied to the container, it retains the root ownership.

A simple workaround is to execute the command `./bin/root chown -R app:app /var/www/.composer/` after the unsuccessful installation and then rerun your one-line install command.
